caluma/caluma_workflow/domain_logic.py

```diff
@@ -0,0 +1,96 @@
+from django.core.exceptions import ValidationError
+
+from caluma.caluma_core.events import send_event
+from caluma.caluma_user.models import BaseUser
+
+from ..caluma_form.models import Document
+from . import events, models, utils
+
+
+class StartCaseLogic:
+    @staticmethod
+    def validate_for_start(data):
+        form = data.get("form")
+        workflow = data.get("workflow")
+
+        if form:
+            if (
+                not workflow.allow_all_forms
+                and not workflow.allow_forms.filter(pk=form.pk).exists()
+            ):
+                raise ValidationError(
+                    f"Workflow {workflow.pk} does not allow to start case with form {form.pk}"
+                )
+
+        return data
+
+    @staticmethod
+    def pre_start(validated_data, user):
+        parent_work_item = validated_data.get("parent_work_item")
+        validated_data["status"] = models.Case.STATUS_RUNNING
+
+        form = validated_data.pop("form", None)
+        if form:
+            validated_data["document"] = Document.objects.create(
+                form=form, created_by_user=user.username, created_by_group=user.group
+            )
+
+        if parent_work_item:
+            case = parent_work_item.case
+            while hasattr(case, "parent_work_item"):
+                case = case.parent_work_item.case
+            validated_data["family"] = case
+
+        return validated_data
+
+    @staticmethod
+    def post_start(case, user, parent_work_item):
```

Comment:
Same here: naming, docstring

Reply:
This is exactly as discussed and agreed yesterday. Also I think I'm the wrong person to write that docstring (which I think is unnecessary) since I just moved the code from the serializer (where it was undocumented) to here.